research/picai/preprocessing_transforms.py

```diff
@@ -0,0 +1,176 @@
+    """
+    Calculate target size for cropping and/or padding input image.
+
+    Adapted from https://github.com/DIAGNijmegen/picai_prep.
+
+    Args:
+        image (sitk.Image): Image to be resized.
+        size (Sequence[int]): Target size in voxels (z, y, x).
+        physical_size: Target size in mm (z, y, x)
+
+    Either size or physical_size must be provided.
+
+    Returns:
+        Tuple[Sequence[int], Sequence[int]]:
+            Shape of original image (in convention of SimpleITK (x, y, z) or numpy (z, y, x)) and
+            Size of target image (in convention of SimpleITK (x, y, z) or numpy (z, y, x))
+    """
+    # input conversion and verification
+    if physical_size is not None:
+        # convert physical size to voxel size (only supported for SimpleITK)
+        if not isinstance(image, sitk.Image):
+            raise ValueError("Crop/padding by physical size is only supported for SimpleITK images.")
+        spacing_zyx = list(image.GetSpacing())[::-1]
+        size_zyx = [length / spacing for length, spacing in zip(physical_size, spacing_zyx)]
+        size_zyx = [int(np.round(x)) for x in size_zyx]
+
+        if size is None:
+            # use physical size
+            size = size_zyx
+        else:
+            # verify size
+            if list(size) != list(size_zyx):
+                raise ValueError(
+                    f"Size and physical size do not match. Size: {size}, physical size: "
+                    f"{physical_size}, spacing: {spacing_zyx}, size_zyx: {size_zyx}."
+                )
+
+    if isinstance(image, sitk.Image):
+        # determine shape and convert convention of (z, y, x) to (x, y, z) for SimpleITK
+        shape = image.GetSize()
+        size = list(size)[::-1]
+    else:
+        # determine shape for numpy array
+        assert isinstance(image, (np.ndarray, np.generic))
+        shape = image.shape
+        size = list(size)
+    rank = len(size)
+    assert (
+        rank <= len(shape) <= rank + 1
+    ), f"Example size doesn't fit image size. Got shape={shape}, output size={size}"
+
+    return shape, size
+
+
+def crop_or_pad(
+    image: sitk.Image,
+    size: Sequence[int] = (20, 256, 256),
+    physical_size: Optional[Sequence[float]] = None,
+    crop_only: bool = False,
+) -> sitk.Image:
+    """
+    Resize image by cropping and/or padding.
+
+    Adapted from https://github.com/DIAGNijmegen/picai_prep.
+
+    Args:
+        image (sitk.Image): Image to be resized.
+        size (Sequence[int]): Target size in voxels (z, y, x).
+        physical_size: Target size in mm (z, y, x)
+
+    Either size or physical_size must be provided.
+
+    Returns:
+        sitk.Image: Cropped or padded image.
+    """
+    # input conversion and verification
+    shape, size = input_verification_crop_or_pad(image, size, physical_size)
+
+    # set identity operations for cropping and padding
+    rank = len(size)
+    padding = [[0, 0] for _ in range(rank)]
```

Comment:
If you just want a list of [0,0] and a list of slice(None) of the same size as rank I think you can just do
```
padding = [[0,0]]*rank
slicer = [slice(None)]*rank
```